device/ospf/ospfv2/interface/auth.yaml

```diff
@@ -0,0 +1,42 @@
+components:
+  schemas:
+    Ospfv2Interface.Authentication:
+      description: |-
+        This contains OSPFv4 authentication properties.
+        Reference: https://www.rfc-editor.org/rfc/rfc2328#appendix-D
+      type: object
+      required: [auth_type]
+      properties:
+        auth_type:
+          description: |-
+            The authentication method.
+            - md5: If the authentication type is of 'md5' then 'md5_key_id' and 'md5_key' both are to be configured.
+            - password: 
+          type: string
+          x-field-uid: 1
+          x-enum:
+            md5:
+              x-field-uid: 1
+            password:
+              x-field-uid: 2
+        md5_key_id:
+          description: |-
+            The unique MD5 Key Identifier per-interface.
+          type: string
+          minLength: 0
+          maxLength: 255
+          x-field-uid: 2          
+        md5_key:
+          description: |-
+            Authentication as an MD5 key.
```

Comment:
Maybe something like :  An alphanumeric secret used to generate the 16 byte MD5 hash value added to the OSPFv2 PDU in the Authentication TLV  ... or something like that.